docs/book/src/reference/scopes.md

```diff
@@ -0,0 +1,115 @@
+# Understanding and Setting Scopes for Managers (Operators) and CRDs
+
+This section covers the configuration of the operational and resource scopes
+within a Kubebuilder project. Managers("Operators") in Kubernetes can be scoped to either
+specific namespaces or the entire cluster, influencing how resources are watched and managed.
+
+Additionally, CustomResourceDefinitions (CRDs) can be defined to be either
+namespace-scoped or cluster-scoped, affecting their availability
+across the cluster.
+
+## Configuring Manager Scope
+
+Managers can operate under different scopes depending on
+the resources they need to handle:
+
+### (Default) Watching All Namespaces
+
+By default, if no namespace is specified, the manager will observe all namespaces.
+This is configured as follows:
+
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+   ...
+})
+```
+
+### Watching a Single Namespace
+
+To constrain the manager to monitor resources within a specific namespace, set the Namespace option:
+
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+    ...
+	Cache: cache.Options{
+        DefaultNamespaces: map[string]cache.Config{"operator-namespace": cache.Config{}},
+    },
+})
+```
+
+### Watching Multiple Namespaces
+
+A manager can also be configured to watch a specified set of namespaces using [Cache Config][CacheConfig]:
+
+```go
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+    ...
+	Cache: cache.Options{
+        DefaultNamespaces: map[string]cache.Config{
+        "operator-namespace1": cache.Config{},
+        "operator-namespace2": cache.Config{},
+        },
+    },
+})
+```
+
+## Configuring CRD Scope
+
+The scope of CRDs determines their visibility either within specific namespaces or across the entire cluster.
+
+### Namespace-scoped CRDs
+
+Namespace-scoped CRDs are suitable when resources need to be isolated to specific namespaces. This setting helps
+manage resources related to particular teams or applications and facilitates phased
+testing and rollout strategies.
```

Comment:
I really like the scope summary, but there is only one thing which is "important" to mention - as i had this discussion multiple times when people start developing operators:

> This setting helps manage resources related to particular teams or applications and facilitates phased
> testing and rollout strategies.

This could be misinterpreted, as the CRD definition (kind + version) is unique from an k8s API server PoV and therefore it's not possible to "test" a new version of the CRD.

And even if the CRDs are having a good versioning and you implement some conversion, you have to decide which manager-instance is responsible for the conversion.

```yaml
spec:
  conversion:
    strategy: Webhook
    webhook:
      clientConfig:
        caBundle: 
        service:
          name: operator-webhook-service
          namespace: operator-system
          path: /convert
      conversionReviewVersions:
      - v1
      - v1beta1
```

It's still possible to do that, but this must be somehow coordinated - which means, the deployable manifests between stages differ essentially.

The namespace scope must be also taken into account in:
* `mutatingwebhookconfiguration`
* `validatingwebhookconfiguration`

Reply:
Changed here: https://github.com/kubernetes-sigs/kubebuilder/pull/4035/files#r1687604796
please, let me know wdyt.